src/main/groovy/com/calgaryscientific/gradle/VeracodeBeginScanTask.groovy

```diff
@@ -35,21 +35,22 @@ class VeracodeBeginScanTask extends VeracodeTask {
     VeracodeBeginScanTask() {
         description = "Begin a Veracode Scan for the given 'app_id'"
         requiredArguments << 'app_id'
-        dependsOn "veracodeGetPreScanResults"
         app_id = project.findProperty("app_id")
         defaultOutputFile = new File("${project.buildDir}/veracode", "build-info-${app_id}-latest.xml")
     }
 
-    VeracodeGetPreScanResultsTask preScan = new VeracodeGetPreScanResultsTask()
-    File preScanResultsOutputFile = preScan.getOutputFile()
+    VeracodeGetPreScanResultsTask getPreScanResultsTask = new VeracodeGetPreScanResultsTask()
 
     Set<String> getModuleWhitelist() {
         veracodeSetup = project.findProperty("veracodeSetup") as VeracodeSetup
         return veracodeSetup.moduleWhitelist
     }
 
     void run() {
-        Set<String> moduleIds = VeracodePreScanResults.extractWhitelistModuleIds(XMLIO.readXml(preScanResultsOutputFile), getModuleWhitelist())
+        getPreScanResultsTask.app_id = app_id
+        getPreScanResultsTask.veracodeAPI = veracodeAPI
+        getPreScanResultsTask.run()
```

Comment:
- there are couple concerns I have with this:
  - this task reach too deep and depends on another gradle task (would it be easier to test if the prescanresults behavior is isolated in domain model instead of gradle tasks?)
  - and this gradle task would call the behavior of that PreScanResults to get the pre-scan results behavior?
  - run(), what does run do? run is the API of the class, it is the method that would be used by other classes, should it be named to something that has explicit/descriptive meaning?